### PR TITLE
REST API for channels

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #29 

#### What's this PR do?
Implements rest API for channels

#### How should this be manually tested?
You can go to `/api/v0/channels/` and play around with it using the DRF HTML view. You should also be able to go to `/api/v0/channels/<subreddit>/` to view a particular channel or to patch it

### CODE REVIEW

File: channels/serializers.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        )
+
+    def update(self, instance, validated_data):
+        api = Api()
+        api.user = True
+
+        name = instance.display_name
+        key_map = {
+            'title': 'title',
+            'channel_type': 'subreddit_type',
```
Comment (rhysyngsun) [inline channels/serializers.py]: The arg for `update_channel` is actually also `channel_type` so this mapping is unnecessary and actually causing updates to `channel_type` to not work. `update_channel validates it's `kwargs` so you could probably just catch the `ValueError` and reraise as a drf `ValidationError`.

Comment (rhysyngsun) [inline channels/serializers.py]: I tried fixing this locally and still didn't see the channel_type update so something else is going on here too.

Comment (noisecapella) [inline channels/serializers.py]: I see a problem too, I'll look into it

**Comment by noisecapella** [comment]
Should be ready for review again. I also rebased on #44

**Pull request merged**